File: day 07/1.cpp

// # ?LEETCODDE
// # 448. Find All Numbers Disappeared in an Array


// # SELF EXPLANATORY

class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        set<int>s(nums.begin(),nums.end());
        vector<int>ans;
        int n=nums.size();
        for(int i=1;i<=n;i++){
            if(s.find(i)==s.end()){
                ans.push_back(i);
            }
        }
        return ans;
    }
};

File: day 07/1.py
# ?LEETCODDE
# 448. Find All Numbers Disappeared in an Array


# SELF EXPLANATORY

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n=len(nums)
        numset=set(nums)

        s=[]
        for i in range(1,n+1):
            if i not in numset:
                s.append(i)
            

        return s

